### PR TITLE
*: refine pipelined dml benchmarks

### PR DESCRIPTION
**Pull request by ekexium**
### What problem does this PR solve?

Issue Number: ref #50215

Problem Summary:

### What changed and how does it work?

Refine some benchmarks to make them more realistic and stable.

### Check List

Tests 

- [ ] Unit test
- [ ] Integration test
- [ ] Manual test (add detailed scripts or steps below)
- [x] No need to test
  > - [ ] I checked and no code files have been changed.
  > 

Side effects

- [ ] Performance regression: Consumes more CPU
- [ ] Performance regression: Consumes more Memory
- [ ] Breaking backward compatibility

Documentation

- [ ] Affects user behaviors
- [ ] Contains syntax changes
- [ ] Contains variable changes
- [ ] Contains experimental features
- [ ] Changes MySQL compatibility

### Release note

Please refer to [Release Notes Language Style Guide](https://pingcap.github.io/tidb-dev-guide/contribute-to-tidb/release-notes-style-guide.html) to write a quality release note.

```release-note
None
```

### CODE REVIEW

File: pkg/session/bench_test.go [hunk 1]
```diff
@@ -1949,6 +1950,8 @@ var batchNum = 100
 var batchSize = 100
 
 func BenchmarkPipelinedSimpleInsert(b *testing.B) {
+	failpoint.Enable("tikvclient/pipelinedSkipResolveLock", "return")
```
Comment (cfzjywxk) [inline pkg/session/bench_test.go:1953]: Does it mean the it's tested without commit the prewrite locks?

Comment (ekexium) [reply pkg/session/bench_test.go:1953]: Yes. Hopefully it will make the test more stable.

**Review by cfzjywxk — APPROVED** [review]
LGTM

**Review by you06 — APPROVED** [review]

File: pkg/session/bench_test.go [hunk 2]
```diff
@@ -2072,6 +2081,8 @@ func BenchmarkPipelinedDelete(b *testing.B) {
 }
 
 func BenchmarkPipelinedReplaceNoDuplicates(b *testing.B) {
+	failpoint.Enable("tikvclient/pipelinedSkipResolveLock", "return")
+	defer failpoint.Disable("tikvclient/pipelinedSkipResolveLock")
```
Comment (hawkingrei) [inline pkg/session/bench_test.go:2085]: You should check the return result to see if a failpoint exists.

**Pull request merged**